website/mailing_list/views.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from website.project.decorators import (
+    must_be_valid_project,
+    must_have_permission,
+    must_not_be_registration,
+)
+from website.util.permissions import ADMIN
+from website.mailing_list import utils
+
+
+###############################################################################
+# View Functions
+###############################################################################
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def get_node_mailing_list(node, auth, **kwargs):
+    return format_node_data_recursive([node], auth.user)
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def enable_mailing_list(node, **kwargs):
+    utils.celery_create_list(node._id)
+    node.mailing_enabled = True
+    node.save()
+    return {'message': 'Successfully enabled mailing lists', 'success': True}
+
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def disable_mailing_list(node, **kwargs):
+    utils.celery_delete_list(node._id)
+    node.mailing_enabled = False
```

Comment:
It seems like there is a situation where this could get out of sync- would you mind walking me through how we handle the following scenario?

1. The user disables the mailing list from the web page.
2. This view immediately marks the list as disabled.
3. The delete operation fails for some unknown reason (eg Mailgun returns a 500 error)
4. Now, the OSF thinks the list is gone... but according to mailgun, the list is still there. (inconsistent state across two sources of record)

Questions:
a. Do we have a mechanism to retry deletion later if it fails?
b. Will users still be able to receive emails from a list that they thought was gone?
c. Are there any other issues that could arise here?